### PR TITLE
#266 Implement admin options for large scale share management

### PR DESCRIPTION
**Pull request by nixnuex**
Enhanced share admin action with intermediate form for bulk share management

### CODE REVIEW

File: juntagrico/admins/forms/admin_mark_share.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+FIELD_CHOICES = [('paid_date', _('Bezahlt am')),
+                 ('issue_date', _('Ausgestellt am')),
+                 ('booking_date', _('Eingebucht am')),
+                 ('cancelled_date', _('Gekündigt am')),
+                 ('termination_date', _('Gekündigt auf')),
+                 ('payback_date', _('Zurückbezahlt am'))]
+
+
+class MarkShareOptionsForm(forms.Form):
```
Comment (motu55) [inline juntagrico/admins/forms/admin_mark_share.py:14]: not sure if mark is the right word here i would use EditShareDatesForm

File: juntagrico/admins/share_admin.py [hunk 2]
```diff
@@ -11,11 +14,30 @@ class ShareAdmin(BaseAdmin):
     search_fields = ['id', 'member__email', 'member__first_name', 'member__last_name', 'number', 'paid_date',
                      'issue_date', 'booking_date', 'cancelled_date', 'termination_date', 'payback_date']
     raw_id_fields = ['member']
-    actions = ['mark_paid']
+    actions = ['mark_share']
```
Comment (motu55) [inline juntagrico/admins/share_admin.py:17]: use mass_edit_share_dates

File: juntagrico/admins/share_admin.py [hunk 3]
```diff
@@ -11,11 +14,30 @@ class ShareAdmin(BaseAdmin):
...
 
-    def mark_paid(self, request, queryset):
-        for share in queryset.all():
-            share.paid_date = share.paid_date or timezone.now().date()
-            share.save()
+    def mark_share(self, request, queryset):
```
Comment (motu55) [inline juntagrico/admins/share_admin.py:19]: same naming here

File: juntagrico/admins/share_admin.py [hunk 4]
```diff
@@ -11,11 +14,30 @@ class ShareAdmin(BaseAdmin):
...
 
-    mark_paid.short_description = _('ausgewählte {} als bezahlt eintragen').format(Config.vocabulary('share_pl'))
+        return render(request,
+                      'admin/mark_share_intermediate.html',
+                      context={
+                          'shares': queryset,
+                          'form': form,
+                      })
+
+    mark_share.short_description = _('Ausgewählte {} bearbeiten').format(Config.vocabulary('share_pl'))
```
Comment (motu55) [inline juntagrico/admins/share_admin.py:43]: Datum bei ausgewählten shae_pl setzen.

File: juntagrico/templates/admin/mark_share_intermediate.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{% extends "admin/base_site.html" %}
+{% load i18n %}
+{% load config %}
+
+{% block content %}
+    {% vocabulary "share" as v_share %}
+    {% vocabulary "share_pl" as v_share_pl %}
+<h1>{% blocktrans %}{{ v_share_pl }} bearbeiten{% endblocktrans %}</h1>
```
Comment (motu55) [inline juntagrico/templates/admin/mark_share_intermediate.html:8]: same here

File: juntagrico/templates/admin/mark_share_intermediate.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    <h2>{% trans "Zusammenfassung" %}</h2>
+    <ul>
+        <li>{{ v_share_pl }}: {{ shares|length }}</li>
+    </ul>
+
+    <h2>{% trans "Objekte" %}</h2>
+    <ul>
+        {% for share in shares %}
+        <li>{{ v_share }}: <a href="{% url 'admin:juntagrico_share_change' share.id %}">{{ share }}</a></li>
+        <input type="hidden" name="_selected_action" value="{{ share.pk }}" />
```
Comment (motu55) [inline juntagrico/templates/admin/mark_share_intermediate.html:25]: wont that not get overwritten?

**Review by motu55 — CHANGES_REQUESTED** [review]
good idea
some naming should be changed otherwise it is ok

File: juntagrico/templates/admin/mark_share_intermediate.html (hunk 6 above)
Comment (nixnuex) [reply juntagrico/templates/admin/mark_share_intermediate.html:25]: This is how to transfer the pks through the action steps. Works as expected.

**Review by motu55 — APPROVED** [review]

**Pull request merged**